src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.OutputSpec;
+import com.google.api.codegen.config.FieldModel;
+import com.google.api.codegen.config.MethodModel;
+import com.google.api.codegen.config.TypeModel;
+import com.google.api.codegen.viewmodel.OutputView;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Collections;
+import java.util.List;
+
+class OutputTransformer {
+  private static final String RESPONSE_PLACEHOLDER = "$resp";
+
+  static List<OutputSpec> defaultOutputSpecs(MethodModel method) {
+    if (method.isOutputTypeEmpty()) {
+      return Collections.emptyList();
+    }
+    return Collections.singletonList(
+        OutputSpec.newBuilder().addPrint("%s").addPrint(RESPONSE_PLACEHOLDER).build());
+  }
+
+  static OutputView toView(OutputSpec config, MethodContext context) {
+    Runnable once =
+        new Runnable() {
+          boolean ran;
+
+          @Override
+          public void run() {
+            if (ran) {
+              throw new IllegalArgumentException("only one field of OutputSpec may be set");
+            }
+            ran = true;
+          }
+        };
+
+    OutputView.Builder view = OutputView.newBuilder();
+    if (config.hasLoop()) {
+      once.run();
+      throw new UnsupportedOperationException("loop not implemented yet");
+    }
+    if (config.getPrintCount() > 0) {
+      once.run();
+      view.print(toView(config.getPrintList(), context)).kind(OutputView.Kind.PRINT);
+    }
+    return view.build();
+  }
+
+  private static OutputView.PrintView toView(List<String> config, MethodContext context) {
+    Preconditions.checkArgument(!config.isEmpty(), "print spec cannot be empty");
+    return OutputView.PrintView.newBuilder()
+        .printSpec(context.getNamer().getPrintSpec(config.get(0)))
+        .printArgs(
+            config
+                .subList(1, config.size())
+                .stream()
+                .map(a -> accessor(a, context))
```

Comment:
It would be helpful to our users if we could also check that the number of placeholders in the input string matches the number of accessors provided.

Reply:
I thought of ways to do this and decided to write a general algorithm for it. Let's do this in a separate PR.

Reply:
I think this could be just a couple of lines in this file, no? Just check the two counts.